dask_jobqueue/flux.py

```diff
@@ -0,0 +1,187 @@
+
+        if walltime is None:
+            walltime = dask.config.get("jobqueue.%s.walltime" % self.config_name)
+        if job_cpu is None:
+            job_cpu = dask.config.get("jobqueue.%s.job-cpu" % self.config_name)
+
+        header_lines = []
+
+        # Error and output files
+        error = "%s/%s-%%J.err" % (self.log_directory, self.job_name or "worker")
+        output = "%s/%s-%%J.out" % (self.log_directory, self.job_name or "worker")
+
+        # Flux header build
+        if self.job_name is not None:
+            header_lines.append("#flux: --job-name=%s" % self.job_name)
+        if self.log_directory is not None:
+            header_lines.append("#flux: --error %s" % error)
+            header_lines.append("#flux: --output %s" % output)
+
+        if queue is not None:
+            header_lines.append("#flux: --queue=%s" % queue)
+
+        # Init resources, always 1 task
+        header_lines.append("#flux: -n 1")
+        header_lines.append("#flux: --cores-per-slot=%s" % self.worker_cores)
+
+        if walltime is not None:
+            header_lines.append("#flux: -t %s" % timestr2seconds(walltime))
+
+        # Skip requested header directives
+        header_lines = list(
+            filter(
+                lambda line: not any(skip in line for skip in self.job_directives_skip),
+                header_lines,
+            )
+        )
+        # Add extra header directives
+        header_lines.extend(["#flux: %s" % arg for arg in self.job_extra_directives])
+
+        # Declare class attribute that shall be overridden
+        self.job_header = "\n".join(header_lines)
+
+    def _job_id_from_submit_output(self, out):
+        """The flux jobid looks like ƒBXsRhMtb"""
+        return out.strip()
+
+    @contextmanager
+    def job_file(self):
+        """Write job submission script to temporary file
+
+        We also want to make this executable and return the full path,
+        and not cleanup at the end, as doing so can create a race condition
+        between flux reading / starting the job and cleaning it up.
+        """
+        fd, path = tempfile.mkstemp(prefix="flux-job-", suffix=".sh")
+        try:
+            logger.debug("writing job script: \n%s", self.job_script())
+            with os.fdopen(fd, "w") as f:
+                f.write(self.job_script())
+        except Exception:
+            raise ValueError("Issue writing file %s" % path)
+
+        # Make the file executable, return the full path
+        st = os.stat(path)
+        os.chmod(path, st.st_mode | stat.S_IEXEC | stat.S_IXGRP | stat.S_IXOTH)
+        yield os.path.abspath(path)
+
+
+class FluxCluster(JobQueueCluster):
+    __doc__ = """Launch Dask on a Flux cluster
+
+    Parameters
+    ----------
+    queue : str
+        Destination queue for each worker job. Passed to `#SBATCH -p` option.
+    project : str
+        Deprecated: use ``account`` instead. This parameter will be removed in a future version.
+    account : str
```

Comment:
Maybe you should just remove the project, account, and job_mem from the docs if it isn't supported by Flux? There was some discussions to move them up, but all Scheduler doesn't take same kwargs, so you should be able to just remove all the related code.